Core/clim-core/panes.lisp

```diff
@@ -2979,33 +2979,43 @@ current background message was set."))
                       borders
                       label))
   (setf port (or port (find-port)))
-  (let* ((fm (find-frame-manager :port port))
-         (frame (make-application-frame 'a-window-stream
-                                        :frame-event-queue input-buffer
-                                        :frame-manager fm
-                                        :pretty-name (or label "")
-					:left left
-					:top top
-					:right right
-					:bottom bottom
-					:width width
-					:height height
-                                        :scroll-bars scroll-bars)))
-    ;; Adopt and enable the pane
-    (when (eq (frame-state frame) :disowned)
-      (adopt-frame fm frame))
-    (unless (or (eq (frame-state frame) :enabled)
-		(eq (frame-state frame) :shrunk))
-      (enable-frame frame))
-    ;; Start a new thread to run the event loop, if necessary.
-    (let ((*application-frame* frame))
-      (stream-set-input-focus (encapsulating-stream-stream frame)))
-    #+clim-mp
-    (unless input-buffer
-      (redisplay-frame-panes frame :force-p t)
-      (clim-sys:make-process (lambda () (let ((*application-frame* frame))
-                                          (standalone-event-loop)))))
-    (encapsulating-stream-stream frame)))
+  ;; Input buffers in the spec are not well defined for panes but at least we
+  ;; know that they are vectors while event queues are deliberely
+  ;; unspecified. OPEN-WINDOW-STREAM description is fudged in this regard by
+  ;; allowing to specify input-buffer as either. -- jd 2019-06-21
+  (multiple-value-bind (event-queue input-buffer)
+      (typecase input-buffer
+        (event-queue (values input-buffer nil))
+        (vector      (values nil input-buffer))
+        (otherwise   (values nil nil)))
+    (let* ((fm (find-frame-manager :port port))
+           (frame (make-application-frame 'a-window-stream
+                                          :event-queue event-queue
+                                          :input-buffer input-buffer
```

Comment:
Is supplying `nil` (instead of nothing) ok for these? At least `make-pane-1` checks for supplied keyword arguments, not for `nil` values.

Reply:
good point, fixed with apply